lms/common/types.go

```diff
@@ -119,10 +119,17 @@ func (x lms_type_code) LmsType() (lms_type_code, error) {
 }
 
 // Returns the expected signature length for an LMS type, given an associated LM-OTS type
-func (x lms_type_code) LmsSigLength(otstc lms_type_code) uint64 {
+func (x lms_type_code) LmsSigLength(otstc lms_type_code) (uint64, error) {
 	if x >= LMS_SHA256_M32_H5 && x <= LMS_SHA256_M32_H25 {
-		params := x.LmsParams()
-		return uint64(4 + 4 + otstc.LmsOtsSigLength() + (params.H * params.M))
+		params, err := x.LmsParams()
+		if err != nil {
+			return 0, err
+		}
+		otssiglen, err := otstc.LmsOtsSigLength()
+		if err != nil {
+			return 0, err
+		}
+		return uint64(4 + 4 + otssiglen + (params.H * params.M)), nil
 	} else {
 		panic("LmsSigLength(): invalid type code")
```

Comment:
There's another panic here -- do we want to remove it as well?

Reply:
Yes.